### PR TITLE
Add dependencies syncronizer utility

### PR DESCRIPTION
**Pull request by sneridagh**
Now you can run it directly from the core repo:

```shell
node ./packages/scripts/volto-update-deps.js path/to/my/project
```

Once released, if you have installed `@plone/scripts:

```shell
pnpm run volto-update-deps path/to/my/project
```
or
```shell
./node_modules/.bin/volto-update-deps.js path/to/my/project
```

It will sync the `dependencies` and `devDependencies` with the ones in Volto. It never removes dependencies (for obvious reasons).

It eases the upgrades by not having to modify manually versions over and over. This will allow to move to `pnpm` everywhere :)

The setup is already being tested with the current setup.

### CODE REVIEW

**Comment by sneridagh** [comment]
Muahahaha next will be green :)

**Review by ichim-david — APPROVED** [review]
@sneridagh this is awesome, my only nitpick is regarding the actual command name
depssync. I would rather have it depsSync, taking the queues from the package.json
dependencies
devDependencies.
It makes it easier to read to have that camelCase naming since it's two words in one

**Comment by ichim-david** [comment]
> @sneridagh this is awesome, my only nitpick is regarding the actual command name depssync. I would rather have it depsSync, taking the queues from the package.json dependencies devDependencies. It makes it easier to read to have that camelCase naming since it's two words in one

@sneridagh taking a better look at my node_modules/.bin folder I see that all of the commands are split by a dash only as such we should do the same
deps-sync just like we have "dry-release".

**Comment by davisagli** [comment]
@sneridagh @ichim-david the bin folder is a flat namespace. I would name it something that makes it clearer the command comes from volto, like `volto-update-deps`

(Really I'd rather have one `volto-dev` command with multiple subcommands, but that is a project for another day.)

File: packages/scripts/README.md [hunk 1]
```diff
@@ -10,6 +10,27 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
 
 This script is installed in the `node_modules/.bin` directory and can be called via `yarn i18n` or directly in the `scripts` `package.json` part.
 
+## depssync
+
+Given an existing project path, it updates the Volto `dependencies` and `devDependencies` for the existing Volto (`@plone/volto`) version in the `package.json` file.
```
Comment (stevepiercy) [inline packages/scripts/README.md:15]:
```suggestion
Given an existing project path, `depssync` updates the Volto `dependencies` and `devDependencies` for the existing Volto (`@plone/volto`) version in the `package.json` file.
```

File: packages/scripts/README.md [hunk 2]
```diff
@@ -10,6 +10,27 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
...
+
+If you are in the process to upgrade your project to a given Volto version, you should update the `package.json` `@plone/volto` dependencies key to use the updated version, then run the script.
```
Comment (stevepiercy) [inline packages/scripts/README.md:17]:
```suggestion
If you are in the process of upgrading your project to a given Volto version, you should update the `package.json` `@plone/volto` dependencies key to use the updated version, then run the script.
```

File: packages/scripts/README.md [hunk 3]
```diff
@@ -10,6 +10,27 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
...
+The script is included in `@plone/scripts` package, you can run it from your project:
```
Comment (stevepiercy) [inline packages/scripts/README.md:18]:
```suggestion
The script is included in the `@plone/scripts` package.
You can run it from inside your project's directory root.
```

File: packages/scripts/README.md [hunk 4]
```diff
@@ -10,6 +10,27 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
...
+
+```shell
+pnpm run volto-update-deps .
+```
+
+or run it directly from the `.bin` folder.
```
Comment (stevepiercy) [inline packages/scripts/README.md:24]:
```suggestion
Or run it directly from the `.bin` folder.
```

File: packages/scripts/README.md [hunk 5]
```diff
@@ -10,6 +10,27 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
...
+
+```shell
+./node_modules/.bin/volto-update-deps.js path/to/my/project
+```
+
+## Deprecated utilites
+
+The following utilities are no longer used by core, nor in projects
```
Comment (stevepiercy) [inline packages/scripts/README.md:32]:
```suggestion
The following utilities are no longer used by core or in projects.
```

Comment (stevepiercy) [reply packages/scripts/README.md:32]: Is that correct, that _everything_ after this sentence is no longer used?

File: packages/scripts/news/5879.feature [hunk 6]
```diff
@@ -0,0 +1 @@
+Project dependencies sync utility `dep_sync` @sneridagh
```
Comment (stevepiercy) [inline packages/scripts/news/5879.feature:1]: We should make sure this name aligns with `depssync`, yes?

```suggestion
Added project dependencies sync utility `depssync`. @sneridagh
```

File: packages/volto/news/5879.breaking [hunk 7]
```diff
@@ -0,0 +1 @@
+Move `devDependencies` and `dependencies` where they belong @sneridagh
```
Comment (stevepiercy) [inline packages/volto/news/5879.breaking:1]:
```suggestion
Moved `devDependencies` and `dependencies` to where they belong. @sneridagh
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: packages/scripts/README.md [hunk 8]
```diff
@@ -10,6 +10,27 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
 
 This script is installed in the `node_modules/.bin` directory and can be called via `yarn i18n` or directly in the `scripts` `package.json` part.
 
+## depssync
```
Comment (stevepiercy) [inline packages/scripts/README.md:13]: One more.

```suggestion
## `volto-update-deps`
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: packages/scripts/README.md (hunk 5 above)
Comment (sneridagh) [reply packages/scripts/README.md:32]: Could be that people still rely on them, so I think it's better left them, they make no harm. 

We can remove them in the next major.

**Comment by sneridagh** [comment]
@davisagli @ichim-david doh, it seems I didn't commited the key file :)

File: packages/scripts/README.md (hunk 5 above)
Comment (stevepiercy) [reply packages/scripts/README.md]: In that case, we should make all the subsequent headings children of this heading by adding the required number of `#`. My question was more about the scope of things and how to format them, not whether to keep or remove them.

File: packages/scripts/README.md [hunk 9]
```diff
@@ -10,6 +10,28 @@ See https://6.docs.plone.org/volto/development/i18n.html for more information.
...
+Or run it directly from the `.bin` folder.
+
+```shell
+./node_modules/.bin/volto-update-deps.js path/to/my/project
+```
+
+## Deprecated utilites
+
+The following utilities are no longer used by core or in projects.
+
```
Comment (stevepiercy) [inline packages/scripts/README.md:34]: All of the following utilities' headings should be increased to be subsections of this section.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: packages/scripts/README.md (hunk 9 above)
Comment (sneridagh) [reply packages/scripts/README.md:34]: Done

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Should `### consolidate` be a child of `### addon`?

If so, that needs to be fixed, too.

Otherwise docs LGTM.

**Review by stevepiercy — APPROVED** [review]

**Pull request merged**